glean-core/rlb/src/private/labeled.rs

```diff
@@ -0,0 +1,389 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use inherent::inherent;
+use std::sync::Arc;
+
+use glean_core::metrics::MetricType;
+use glean_core::ErrorType;
+
+use crate::dispatcher;
+
+/// Sealed traits protect against downstream implementations.
+///
+/// We wrap it in a private module that is inaccessible outside of this module.
+mod private {
+    use crate::{
+        private::BooleanMetric, private::CounterMetric, private::StringMetric, CommonMetricData,
+    };
+    use std::sync::Arc;
+
+    /// The sealed labeled trait.
+    ///
+    /// This also allows us to hide methods, that are only used internally
+    /// and should not be visible to users of the object implementing the
+    /// `Labeled<T>` trait.
+    pub trait ToLabeled {
+        /// The `glean_core` metric type representing the labeled metric.
+        type InnerCoreMetric: glean_core::metrics::MetricType + Clone;
```

Comment:
Can't we have this just be `type Inner`? that it is _core_ and that it is a _metric_ is implied by the bound on  `glean_core::metrics::MetricType`.